plugins/modules/aci_l4l7_device_selection_policy.py

```diff
@@ -0,0 +1,314 @@
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        contract=dict(type="str", aliases=["contract_name"]),
+        graph=dict(type="str", aliases=["service_graph", "service_graph_name"]),
+        node=dict(type="str", aliases=["node_name"]),
+        device=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "contract", "graph", "node"]],
+            ["state", "present", ["tenant", "contract", "graph", "node"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    state = module.params.get("state")
+    contract = module.params.get("contract")
+    graph = module.params.get("graph")
+    node = module.params.get("node")
+    device = module.params.get("device")
+
+    policy_dn = "ldevCtx-c-{0}-g-{1}-n-{2}".format(contract, graph, node) if (contract, graph, node) != (None, None, None) else None
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="vnsLDevCtx",
+            aci_rn=policy_dn,
+            module_object=policy_dn,
+            target_filter={"dn": policy_dn},
+        ),
+        child_classes=["vnsRsLDevCtxToLDev"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        if device is not None:
+            device_tdn = "uni/tn-{0}/lDevVip-{1}".format(tenant, device)
+            child_configs.append({"vnsRsLDevCtxToLDev": {"attributes": {"tDn": device_tdn}}})
```

Comment:
scenario never tested please add tests

Reply:
added device binding creation and removal to integration tests